src/shiv/builder.py

```diff
@@ -46,6 +50,34 @@ def write_file_prefix(f: IO[Any], interpreter: str) -> None:
     f.write(b"#!" + interpreter.encode(sys.getfilesystemencoding()) + b"\n")
 
 
+def _write_to_zip_app(
+    zfile: zipfile.ZipFile, arcname: str, data_source: Union[Path, bytes],
+    date_time: Tuple[int, int, int, int, int, int], compression: int,
+    contents_hash: Optional[hashlib.sha3_256] = None,
+) -> None:
```

Comment:
Refactored the signature.
Unfortunately, we cannot remove the date_time here, because we won't be able to make archive reproducible without it. If the user will provide a timestamp value in env variable it will be the only way we can know about it.